docs/Advanced-Topics-Custom-Block-Render.md

```diff
@@ -0,0 +1,152 @@
+|     `<li/>`     | unordered-list-item,ordered-list-item** |
+|    `<div/>`     |               unstyled***               |
+
+\*\* - Block type will be based on the parent `<ul/>` or `<ol/>`
+
+\*\*\* -  Any block that is not recognized by the block redering mapping will be treated as unstyled
+
+## Configuring block render map
+
+Draft default block render map can be overwritten, by passing an
+[Immutable Map](http://facebook.github.io/immutable-js/docs/#/Map) to
+the editor blockRender props.
+
+*example of overwritting default block render map:*
+
+```js
+// The example below deliberatly only allows
+// 'heading-two' as the only valid block type and
+// updates the unstyled element to also become a h2.
+const blockRenderMap = Immutable.Map({
+  'header-two': {
+   element: 'h2'
+  },
+  'unstyled': {
+    element: 'h2'
+  }
+});
+
+class RichEditor extends React.Component {
+  render() {
+    return (
+      <Editor
+        ...
+        blockRenderMap={blockRenderMap}
+      />
+    );
+  }
+}
+```
+
+There are cases where instead of overwriting the defaults we just want to add new block types;
+this can be done by using the DefaultDraftBlockRenderMap reference to create a new blockRenderMap
+
+*example of extending default block render map:*
+
+```js
+const blockRenderMap = Immutable.Map({
+  'section': {
+    element: 'section'
+  }
+});
+
+// Include 'paragraph' as a valid block and updated the unstyled element but
+// keep support for other draft default block types
+const extendedBlockRenderMap = Draft.DefaultDraftBlockRenderMap.merge(blockRenderMap);
+
+class RichEditor extends React.Component {
+  render() {
+    return (
+      <Editor
+        ...
+        blockRenderMap={extendedBlockRenderMap}
+      />
+    );
+  }
+}
+```
+
+## Custom block wrappers
+
+By default the html element is used to wrap block types however a react component
+can also be provided to the _blockRenderMap_ to wrap the EditorBlock.
+
+During pasting or when using the
+[convertFromHTML](https://facebook.github.io/draft-js/docs/api-reference-data-conversion.html#convertfromhtml)
+the html will be scanned for matching tag elements. A wrapper will be used when there is a definition for
+it on the _blockRenderMap_ to wrap that particular block type. For example:
+
+Draft uses wrappers to wrap `<li/>` inside either `<ol/>` or `<ul/>` but wrappers can also be used
+to wrap any other custom block type
+
+*example of extending default block render map to use a react component for a custom block:*
+
+```js
+class MyCustomBlock extends React.Component {
+  constructor(props) {
+    super(props);
+  }
+
+  render() {
+    return (
+      <div className='MyCustomBlock'>
+        {this.props.children}
+      </div>
+    );
+  }
+}
+
+const blockRenderMap = Immutable.Map({
+  'MyCustomBlock': {
+    element: 'section', // will be used during paste or html conversion to auto match your component
+    wrapper: <MyCustomBlock {...this.props} />
```

Comment:
Is this a mistake, I think it would cause a runtime error? What does the `this` refer to in this context?